### PR TITLE
Extend-wasm-ttl

### LINKED ISSUE #618: Create a guide for extending a deployed contract’s Wasm code TTL with code
**Get awarded:** up to $150 worth of XLM
**Section:** How-To Guides - Conventions: https://github.com/stellar/stellar-docs/tree/main/docs/smart-contracts/guides/conventions

**Guide could include:**

- Code walkthrough
- Written explanation

### PR DESCRIPTION
**Pull request by Myestery**
fix #618

### CODE REVIEW

**Comment by Myestery** [comment]
@briwylde08 This is ready for review

**Comment by briwylde08** [comment]
Thanks for the submission! We'll get it assigned and reviewed. Note that reviews may take longer than normal as we have a lot of submissions to get through!

**Comment by briwylde08** [comment]
Hey @Myestery! Sorry for the delay on the review on this. Let me check!

File: docs/build/guides/conventions/extending-wasm-ttl.mdx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+- Get the contract's footprint,
+- Set a reasonable resource fee (perhaps start at 10,000 stroops as this is a costly operation),
+- Set Soroban data as read-only,
+- Set the desired ledger TTL to expire the contract,
+- Create an operation `StellarSdk.Operation.extendFootprintTtl`,
+- Note that a resource fee and base fee are both charged in this operation.
+
+## JavaScript code
+
+The code below uses Nodejs environment but same concept can also be applied in the browser using Freighter wallet or using any other [Stellar SDK](/docs/tools/sdks/library).
```
Comment (ElliotFriend) [inline docs/build/guides/conventions/extending-wasm-ttl.mdx:31]:
```suggestion
The code below uses Nodejs environment but same concept can also be applied in the browser using the Freighter wallet or using any other [Stellar SDK](../../../tools/sdks/library.mdx).
```

File: docs/build/guides/conventions/extending-wasm-ttl.mdx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+);
+
+extendContractWasmTTL(contractId, sourceKeypair)
+  .then(console.log)
+  .catch(console.error);
+```
+
+In the above code, we first import the required modules and define the `extendContractWasmTTL` function. This function takes the contract ID and the source keypair as arguments. It then creates a new transaction builder and fetches the contract's Wasm hash. Next, it sets the Soroban data and creates an operation to extend the contract's TTL. Finally, it signs and submits the transaction.
+
+Read more about the Extend Footprint TTL operation [here](https://developers.stellar.org/docs/learn/encyclopedia/storage/state-archival#extendfootprintttlop).
```
Comment (ElliotFriend) [inline docs/build/guides/conventions/extending-wasm-ttl.mdx:91]:
```suggestion
Read more about the Extend Footprint TTL operation [here](../../../learn/encyclopedia/storage/state-archival.mdx#extendfootprintttlop).
```

File: docs/build/guides/conventions/extending-wasm-ttl.mdx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+  // Get the contract instance
+  const contract = new StellarSdk.Contract(contractId);
+  const instance = contract.getFootprint();
+
+  // Set the Soroban data and create an operation to extend the contract's TTL
+  const sorobanData = new StellarSdk.SorobanDataBuilder()
+    .setResourceFee(200_000)
+    .setReadOnly([instance])
+    .build();
```
Comment (ElliotFriend) [inline docs/build/guides/conventions/extending-wasm-ttl.mdx:53]: I think it's worthwhile here to describe in a little more detail what the `SorobanDataBuilder()` is doing, and _how_ you're using it. Explain why you're using `setReadOnly()` as opposed to `setReadWrite()`. That kind of stuff.

**Review by ElliotFriend — CHANGES_REQUESTED** [review]
Hi, there @Myestery! This is looking really good! A nice, simple, straight-forward guide. I left a couple minor suggestions for improved markdown file links. I'd also like to see just a bit more information/detail about the `SorobanDataBuilder()` part.

Other than that, this one's nearly ready to go! Thanks for the contribution!

**Review by ElliotFriend — APPROVED** [review]
This is wonderful! Thanks for the contribution, @Myestery!

**Comment by briwylde08** [comment]
Thank you @Myestery! As usual, we'll take the award conversation back to the corresponding issue!

**Pull request merged**